modules/core/src/main/java/org/apache/ignite/internal/processors/query/QueryUtils.java

```diff
@@ -1823,4 +1832,64 @@ public KeyOrValProperty(boolean key, String name, Class<?> cls) {
             return -1;
         }
     }
+
+    /** */
+    private enum TemporalTypes {
+        /** Corresponds to SQL DATE. */
+        DATE(java.sql.Date.class, java.time.LocalDate.class),
+
+        /** Corresponds to SQL TIME. */
+        TIME(java.sql.Time.class, java.time.LocalTime.class),
+
+        /** Corresponds to SQL TIMESTAMP. */
+        TIMESTAMP(java.sql.Timestamp.class, java.util.Date.class, java.time.LocalDateTime.class),
+
+        /** Corresponds to non-temporal types. */
+        NONE();
+
+        /** */
+        private final Set<Class<?>> clsSet = new HashSet<>();
+
+        /** */
+        TemporalTypes(Class<?>... types) {
+            if (types == null)
+                return;
+
+            Collections.addAll(clsSet, types);
+        }
+
+        /** */
+        public static TemporalTypes getKind(Class<?> type) {
+            if (DATE.contains(type))
```

Comment:
Why do we need it additional class if we still use `if/elseif`? It will be simplier and faster to just inline the same if/endif into `isConvertibleTypes` method